pandas/core/generic.py

```diff
@@ -11274,9 +11274,9 @@ def _shift_with_freq(self, periods: int, axis: int, freq) -> Self:
             if freq != orig_freq:
                 assert orig_freq is not None  # for mypy
                 raise ValueError(
-                    f"Given freq {freq_to_period_freqstr(freq.n, freq.name)} "
+                    f"Given freq {PeriodDtype(freq)._freqstr} "
```

Comment:
does this change the message? if so, could you show before vs after please?

if not, then is it possible to do this replacement across the whole codebase and get rid of `freq_to_period_freqstr`?

Reply:
> does this change the message? if so, could you show before vs after please?
> 
It doesn't change the message.
> if not, then is it possible to do this replacement across the whole codebase and get rid of `freq_to_period_freqstr`?
>
I replaced `freq_to_period_freqstr` with `PeriodDtype(offset)._freqstr` where it works. Now I am trying to do this replacement in `pandas/tseries/frequencies.py` and in `pandas/_libs/tslibs/period.pyx`, but I get failures. I am not sure we can complitely rid of `freq_to_period_freqstr`